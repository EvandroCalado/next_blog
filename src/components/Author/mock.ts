export default {
  title: 'Evandro Calado',
  resume: 'Desenvolvedor Frontend',
  image: 'https://avatars.githubusercontent.com/u/110628201?v=4',
  alt: 'Evandro Calado',
};
