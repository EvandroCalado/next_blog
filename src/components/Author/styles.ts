import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 3rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border: 3px solid ${theme.colors.secondary};
    border-radius: 50%;
  `}
`;

export const Container = styled.div`
`;

export const Title = styled.h2`
${({ theme }) => css`
font-size: ${theme.font.sizes.xlarge};
`}
`;

export const Paragraph = styled.p`
`;
