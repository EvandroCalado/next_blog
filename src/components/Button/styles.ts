import styled, { css } from 'styled-components';
import { ButtonProps } from './';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background-color: ${theme.colors[color]};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    border: none;
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    margin: ${theme.spacings.large} 0;
    border-radius: 0.5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 300ms ease-in-out;
    cursor: pointer;

    &:hover {
      filter: brightness(80%);
    }

    &:disabled {
      background-color: ${theme.colors.mediumGray};
      color: ${theme.colors.darkGray};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 3rem;
      margin-left: 1rem;
    }
  `}
`;
