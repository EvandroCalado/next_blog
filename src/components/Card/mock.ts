export default {
  slug: 'slug-test',
  title: 'Title',
  cover:
    'https://images.unsplash.com/photo-1593720219276-0b1eacd0aef4?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=640&q=80',
  excerpt:
    'JavaScript é uma linguagem de programação popular e amplamente utilizada para criar e desenvolver aplicativos web dinâmicos e interativos. Ela é interpretada pelo navegador do usuário e é executada em tempo real, permitindo que os desenvolvedores adicionem funcionalidades avançadas aos sites, como animações, interatividade do usuário, validação de formulários, solicitações assíncronas ao servidor e muito mais.',
};
