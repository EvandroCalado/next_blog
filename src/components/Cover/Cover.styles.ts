import styled, { css } from 'styled-components';

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
