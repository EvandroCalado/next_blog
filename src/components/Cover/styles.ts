import styled, { css } from 'styled-components';

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    aspect-ratio: 16/9;
    object-fit: cover;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
