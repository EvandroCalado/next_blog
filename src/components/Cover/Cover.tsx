import * as Styled from './Cover.styles';

export type CoverProps = {
  url: string;
  alt: string;
};

const Cover = ({ url, alt }: CoverProps) => {
  return <Styled.Image src={url} alt={alt} />;
};

export default Cover;
