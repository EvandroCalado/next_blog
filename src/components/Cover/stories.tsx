import Cover, { CoverProps } from '.';
import { Meta, StoryFn } from '@storybook/react';

import mock from './mock';

export default {
  title: 'Cover',
  component: Cover,
  args: mock,
} as Meta;

export const Template: StoryFn<CoverProps> = (args) => (
  <div>
    <Cover {...args} />
  </div>
);
