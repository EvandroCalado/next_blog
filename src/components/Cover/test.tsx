import Cover from './';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

import mock from './mock';

describe('<Cover />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Cover {...mock} />);

    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByRole('img')).toHaveAttribute('src', mock.url);
    expect(screen.getByRole('img')).toHaveAttribute('alt', mock.alt);
    expect(container).toMatchSnapshot();
  });
});
