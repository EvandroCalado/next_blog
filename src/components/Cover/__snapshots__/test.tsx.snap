// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cover /> should render 1`] = `
.c0 {
  max-width: 100%;
  aspect-ratio: 16/9;
  object-fit: cover;
  margin-bottom: 2.4rem;
}

<div>
  <img
    alt="Javascript"
    class="c0"
    src="https://images.unsplash.com/photo-1593720219276-0b1eacd0aef4?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=640&q=80"
  />
</div>
`;
